### PR TITLE
[DOC][#3616] Update the S3 Gateway docs

### PR DESCRIPTION
**Pull request by svekars**

### CODE REVIEW

File: doc/enterprise/s3gateway.md [hunk 1]
```diff
@@ -1,47 +1,252 @@
...
+
+1. Check that the file was added:
+
+   ```bash
+   $ mc ls local/master.raw_data
+   [2019-07-19 12:11:37 PDT]  2.6MiB github_issues_medium.csv
+   [2019-07-19 12:11:37 PDT]     62B test.csv
+   ```
+
+1. Download a file to MinIO from a Pachyderm repository by running:
```
[the remark this answers is not in the document]
Comment (svekars) [reply doc/enterprise/s3gateway.md]: Downloading to MinIO from a Pachyderm repo.

**Comment by svekars** [comment]
Ready for you to review @JoeyZwicker.

**Comment by svekars** [comment]
@JoeyZwicker I think you are right, and I need to fix that part because I got confused myself at first.

File: doc/enterprise/s3gateway.md [hunk 2]
```diff
@@ -1,47 +1,551 @@
...
+   **Example:**
+
+   ```bash
+
+   New settings:
+     Access Key: ""
+     Secret Key: ""
+     Default Region: US
+     S3 Endpoint: localhost:30600
+     DNS-style bucket+hostname:port template for accessing a bucket: 30600/%(bucket)
```
Comment (pappasilenus) [inline doc/enterprise/s3gateway.md]: DNS-style bucket+hostname:port template for accessing a bucket: localhost:30600/%(bucket)
Write

Comment (pappasilenus) [reply doc/enterprise/s3gateway.md]: Somehow dropped the "localhost" here...

File: doc/enterprise/s3gateway.md [hunk 3]
```diff
@@ -204,7 +204,7 @@ The Pachyderm S3 gateway supports the following operations:
 * Get objects: Gets file contents on the HEAD of a branch.
 
 You can use any S3 compatible tool, such as MinIO, AWS CLI, or
-`S3cmd` to interact with the Pachyderm S3 gateway.
```
Comment (ysimonson) [inline doc/enterprise/s3gateway.md]: s3cmd should be all lowercase

Comment (svekars) [reply doc/enterprise/s3gateway.md]: Actually, I think it is S3cmd, but also `s3cmd` according to these:

- https://s3tools.org/s3cmd
- https://github.com/s3tools/s3cmd

**Pull request merged**